homeassistant/components/rfxtrx/__init__.py

```diff
@@ -157,16 +161,48 @@ async def async_setup(hass, config):
 
 async def async_setup_entry(hass, entry: config_entries.ConfigEntry):
     """Set up the RFXtrx component."""
+    hass.data.setdefault(DOMAIN, {})
+
     await hass.async_add_executor_job(setup_internal, hass, entry.data)
 
-    for domain in ["switch", "sensor", "light", "binary_sensor", "cover"]:
+    for domain in DOMAINS:
         hass.async_create_task(
             hass.config_entries.async_forward_entry_setup(entry, domain)
         )
 
     return True
 
 
+async def async_unload_entry(hass, entry: config_entries.ConfigEntry):
+    """Unload RFXtrx component."""
+    await hass.async_add_executor_job(unload_internal, hass, entry.data)
+
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in DOMAINS
+            ]
+        )
+    )
```

Comment:
Missed this. You must let components unload first before you call unload_internal. They could be busy with something.

Reply:
Ok